app/models/backup.rb

```diff
@@ -50,8 +50,23 @@ def s3
 
   def upload_to_s3
     return unless s3
-    File.open(@path) do |file|
-      s3.upload(file, @filename)
+
+    if SiteSetting.s3_enable_encryption
+      if (s3_gnupg_public_key = SiteSetting.s3_gnupg_public_key).blank?
+        raise Discourse::SiteSettingMissing.new("s3_gnupg_public_key")
+      end
+
+      encrypted_filename = "#{@filename}.gpg"
+      encrypted_path = "/tmp/#{encrypted_filename}"
+
+      `echo '#{s3_gnupg_public_key}' | gpg --import`
+      gnupg_user_id = `echo '#{s3_gnupg_public_key}' | gpg --list-keys --with-colons | awk -F: '/^pub:/ { print $5 }'`.chomp
```

Comment:
@mpalmer Can I get your thoughts on this? This is very hacky to me but I'm not quite sure if there are other ways around this. Thank you!